src/meta/MetaVersionMan.cpp

```diff
@@ -15,66 +15,49 @@ DEFINE_bool(print_info, false, "enable to print the rewrite data");
 
 namespace nebula {
 namespace meta {
-static const int32_t metaVersion = 2;
+
 static const std::string kMetaVersionKey = "__meta_version__";          // NOLINT
+
 // static
-int32_t MetaVersionMan::getMetaVersionFromKV(kvstore::KVStore* kv) {
+MetaVersion MetaVersionMan::getMetaVersionFromKV(kvstore::KVStore* kv) {
     CHECK_NOTNULL(kv);
     std::string value;
     auto code = kv->get(kDefaultSpaceId, kDefaultPartId, kMetaVersionKey, &value, true);
-    if (code == kvstore::ResultCode::ERR_KEY_NOT_FOUND) {
-        LOG(INFO) << "There is no meta version key existed in kvstore!";
-        if (isV1(kv)) {
-            return 1;
-        } else {
-            auto ret = setMetaVersionToKV(kv);
-            if (!ret) {
-                return -1;
-            }
-            return metaVersion;
-        }
-    } else if (code == kvstore::ResultCode::SUCCEEDED) {
-        if (value.size() != sizeof(int32_t)) {
-            LOG(ERROR) << "Bad format version " << value;
-            return -1;
-        }
-        auto version = *reinterpret_cast<const int32_t*>(value.data());
-        if (version != metaVersion) {
-            LOG(ERROR) << "Read the version from kv is wrong: " << version;
-            return -1;
-        }
-        return version;
+    if (code == kvstore::ResultCode::SUCCEEDED) {
+        auto version = *reinterpret_cast<const MetaVersion*>(value.data());
+        return version == MetaVersion::V2 ? MetaVersion::V2 : MetaVersion::UNKNOWN;
     } else {
-        LOG(ERROR) << "Error in kvstore, err " << static_cast<int32_t>(code);
-        return -1;
+        return getVersionByHost(kv);
     }
 }
 
 // static
-bool MetaVersionMan::isV1(kvstore::KVStore* kv) {
+MetaVersion MetaVersionMan::getVersionByHost(kvstore::KVStore* kv) {
     const auto& hostPrefix = nebula::meta::MetaServiceUtils::hostPrefix();
     std::unique_ptr<nebula::kvstore::KVIterator> iter;
-    auto code = kv->prefix(kDefaultSpaceId, kDefaultPartId, hostPrefix, &iter);
+    auto code = kv->prefix(kDefaultSpaceId, kDefaultPartId, hostPrefix, &iter, true);
     if (code != kvstore::ResultCode::SUCCEEDED) {
-        return false;
+        return MetaVersion::UNKNOWN;
     }
     if (iter->valid()) {
         auto v1KeySize = hostPrefix.size() + sizeof(int64_t);
-        if (iter->key().size() != v1KeySize) {
-            return false;
+        if (iter->key().size() == v1KeySize) {
+            return MetaVersion::V1;
+        } else {
+            return MetaVersion::V2;
```

Comment:
A little confusion here, is better as below ?
`return (iter->key().size() == v1KeySize) ? MetaVersion::V1 : MetaVersion::UNKNOWN;`

Reply:
Make sense.